protocols/floodsub/src/lib.rs

```diff
@@ -33,3 +35,22 @@ mod rpc_proto {
 pub use self::layer::{Floodsub, FloodsubEvent};
 pub use self::protocol::{FloodsubMessage, FloodsubRpc};
 pub use self::topic::Topic;
+
+/// Configuration options for the Floodsub protocol.
+pub struct FloodsubOptions {
```

Comment:
Could you please name this "config", for consistency with the rest of the library?
```suggestion
pub struct FloodsubConfig {
```

Reply:
I went with "options" because the `libp2p_floodsub::protocol::FloodsubConfig` exists already for protocol (upgrade).. I don't have strong opinion on this naming, but having two of the same name sounds odd. Could you confirm this should be `FloodsubConfig`?

Reply:
Just found https://github.com/libp2p/rust-libp2p/pull/1015 which does propose the `libp2p_floodsub::FloodsubConfig` and renaming the exiting `FloodsubConfig` to `FloodsubListen`... Should I try to bring up the #1015 first and then extend with this behavioural change?

Reply:
#1015 is extremely old, but yes in my opinion we should rename `protocol::FloodsubConfig` to something else.